### DISCUSSION TITLE
Automatic install of Plugins from ZIP

### DISCUSSION THREAD

**Discussion by lars18th** [doublecmd/doublecmd#1757] [Suggestions]
Hi,

In Total Commander, when you try to open a ZIP file with one plugin a new Window informs for it. And if you accept it installes the plugin automatically (inside the correct folder, and to the corresponding type).

I suggest to copy this functionality.
Thank you.

**Comment by Skif-off** [comment]
Automatic installation is implemented to some extent, see [cm_AddPlugin](https://doublecmd.github.io/doc/en/cmds.html#cm_AddPlugin).

**Comment by lars18th** [comment]
Hi @Skif-off ,

Thank you for the response. I don't know about this command. However: _This command is not included in any menu by default._ So I think many users do not know how to use it. Also, it would be easier to automatically trigger a call to this plugin when opening an archive (ZIP) or directory with the INF file. That's the way TC follows.

So, if the main functionality is already implemented... why not include the autodection?

I hope the developers will consider this _small_ enhancement.
Thank you.

**Comment by Skif-off** [comment]
> this _small_ enhancement.

Perhaps a _small_ enhancement does not necessarily require a _small_ effort, I don't know.
I don't remember where this was already discussed. Let's wait for the developer or one of the other programmers.

P.S. I'm not sure if Total Commander and Double Commander can be compared directly here because Total Commander has built-in Zip support and 7-Zip and UnRAR libraries as an integral part, but Double Commander cannot work with archives without plugins or external archivers.

**Reply by lars18th** [reply]
Hi @Skif-off ,

> > this _small_ enhancement.
> 
> Perhaps a _small_ enhancement does not necessarily require a _small_ effort, I don't know. I don't remember where this was already discussed. Let's wait for the developer or one of the other programmers.

IMHO the necessary _effort_ when the functionality is already implemented is to write the code for the UI. But is true that the current `cm_AddPlugin` is very hidden to regular users. Therefore, this needs _some_ improvement.

> P.S. I'm not sure if Total Commander and Double Commander can be compared directly here because Total Commander has built-in Zip support and 7-Zip and UnRAR libraries as an integral part, but Double Commander cannot work with archives without plugins or external archivers.

From my point of view, both are comparable. And Double Commander is superior because the open source nature and the multiplatform support. But it's true that TC has a long time of development, and it has a very useful functionalities that at time are missing in DC.

**Reply by Skif-off** [reply]
> when the functionality is already implemented

There is a different principle here.

> But is true that the current `cm_AddPlugin` is very hidden to regular users.

Not only this :))
(Automatic installation is mentioned [here](https://doublecmd.github.io/doc/en/configuration.html#ConfigPlugins) and [here](https://doublecmd.github.io/doc/en/configuration.html#ConfigDC).)

**Reply by lars18th** [reply]
OK. But when migrating from TC to DC we lose the option to _automatically_ launch the install process when opening the ZIP file. This is the common procedure to install a plugin in TC. Why not support it then in DC?
